lib/improver/tests/utilities/test_pad_spatial.py

```diff
@@ -0,0 +1,367 @@
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the convert_distance_into_number_of_grid_cells function from
+ spatial.py."""
+
+import unittest
+import numpy as np
+import iris
+from iris.coords import DimCoord
+from iris.tests import IrisTest
+
+from improver.utilities.pad_spatial import (
+    pad_coord, pad_cube_with_halo, remove_halo_from_cube,
+    _create_cube_with_padded_data)
+from improver.tests.set_up_test_cubes import set_up_variable_cube
+
+
+class Test_pad_coord(IrisTest):
+
+    """Test the padding of a coordinate."""
+
+    def setUp(self):
+        """Set up a cube."""
+        self.cube = set_up_variable_cube(np.ones((5, 5), dtype=np.float32),
+                                         spatial_grid="equalarea")
+        coord_points_x = np.linspace(10., 50., 5)
+        x_bounds = np.array([coord_points_x - 5, coord_points_x + 5]).T
+        coord_points_y = np.linspace(5., 85., 5)
+        y_bounds = np.array([coord_points_y - 10, coord_points_y + 10]).T
+
+        self.cube.coord("projection_x_coordinate").points = coord_points_x
+        self.cube.coord("projection_x_coordinate").bounds = x_bounds
+        self.cube.coord("projection_y_coordinate").points = coord_points_y
+        self.cube.coord("projection_y_coordinate").bounds = y_bounds
+
+        self.cube_y_reorder = self.cube.copy()
+        coord_points_y_reorder = np.flip(coord_points_y)
+        y_bounds_reorder = np.array([coord_points_y_reorder + 10,
+                                     coord_points_y_reorder - 10]).T
+        self.cube_y_reorder.coord("projection_y_coordinate").points = (
+            coord_points_y_reorder)
+        self.cube_y_reorder.coord("projection_y_coordinate").bounds = (
+            y_bounds_reorder)
+
+    def test_add(self):
+        """Test the functionality to add padding to the chosen coordinate.
+        Includes a test that the coordinate bounds array is modified to reflect
+        the new values."""
+        expected = np.linspace(0., 60., 7)
+        coord = self.cube.coord("projection_x_coordinate")
+        expected_bounds = np.array([expected - 5, expected + 5]).T
+        width = 1
```

Comment:
The original expected values could of been kept if width=2 had been used instead.

Reply:
This is true!  I did this everywhere else and can't remember why I didn't do it here.  Is it I leave this as is (since I think it's fairly clear what the test is doing) or would you like me to change it back?

Reply:
Since it all works we might as well leave as is. It just would of made reviewing easier :-)